src/Tool/Commands/BaseCommand.cs

```diff
@@ -9,25 +11,51 @@
 abstract class BaseCommand
 {
     readonly string[] maskNames;
-    readonly string outputPath;
+    readonly string reportName = "throughput-report";
     readonly bool isDevelopment;
 
     public BaseCommand(string[] maskNames)
     {
         this.maskNames = maskNames;
 
-        outputPath = Path.Combine(Environment.CurrentDirectory, "throughput-report.json");
-
         var envVars = Environment.GetEnvironmentVariables().Keys.OfType<string>().OrderBy(x => x).ToArray();
 
         if (!bool.TryParse(Environment.GetEnvironmentVariable("IS_DEVELOPMENT"), out isDevelopment))
         {
             isDevelopment = false;
         }
+    }
 
+    string GenerateReportTimeStamp(DateTime dateTime)
+    {
+        return $"{dateTime.Year}{dateTime.Month}-{dateTime.Day}-{dateTime.Hour}-{dateTime.Minute}-{dateTime.Second}";
     }
 
-    public async Task Run(CancellationToken cancellationToken = default)
+    bool ReportsExist(string customerName)
+    {
+        var currentDirectory = new DirectoryInfo(Environment.CurrentDirectory);
+
+        var customerReport = $"{customerName}-{reportName}-{GenerateReportTimeStamp(DateTime.Now)}.json";
+
+        var files = currentDirectory.GetFiles($"{customerReport}*");
+
+        return files.Any();
+    }
+
+    bool ConfirmCreateNewReport()
+    {
+        return Confirm($"Existing throughput reports were found.{Environment.NewLine}Would you like to create a new report?");
+    }
+
+    string CreateReportOutputPath(string customerName)
+    {
+        var outputPath = Path.Join(Environment.CurrentDirectory,
+            $"{customerName}-{reportName}");
```

Comment:
Guard against illegal filename chars
```suggestion
        var customerFileName = Regex.Replace(customerName, @"[^\w\d]+", "-").Trim('-').ToLower()  
        var outputPath = Path.Join(Environment.CurrentDirectory,
            $"{customerFileName-{reportName}");
```